app/particle/particle_simulation.py

```diff
@@ -78,58 +79,94 @@
 params.init_total_energy = 0.0
 params.curr_total_energy = 0.0
 
+def add_one_particle(particle_id):
+    radius = 30#random.randint(1, params.max_radius)
+    position = [random.randint(radius, params.simulation_width - 1),
+                random.randint(radius, params.simulation_height - 1),
+                random.randint(radius, params.simulation_depth - 1)]
+    velocity = [400*random.randint(0,radius//4),
+                400*random.randint(0,radius//4),
+                400*random.randint(0,radius//4)]
+    mass = 3#random.randint(1,10)
+    params.init_total_energy += 0.5 * mass * (velocity[0]**2 + velocity[1]**2 + velocity[2]**2)
+    thread_num = util.determine_particle_thread_num(position[0])
+    new_particle = Particle(particle_id, thread_num, position, velocity, mass, radius)
+    params.partitions[thread_num].add_particle(new_particle)
+
+def subtract_one_particle(particle_id):
+    if params.rank is 0:
+        for i in range(1, params.num_active_workers+1):
+            new_particles = params.comm.recv(source = mpi.ANY_SOURCE, status = params.mpi_status, tag = 0)
+            params.partitions[params.mpi_status.Get_source()].particles = new_particles
+        params.init_total_energy -= params.comm.recv(source = mpi.ANY_SOURCE, status = params.mpi_status, tag = 88)
+    elif params.rank <= params.num_active_workers:
+        partition = params.partitions[params.rank]
+        particles = partition.particles
+#        particle_to_be_removed = None
```

Comment:
Code that is commented out can be removed; everything should be in version control